pkg/apis/options/legacy_options_test.go

```diff
@@ -759,4 +763,87 @@ var _ = Describe("Legacy Options", func() {
 			}),
 		)
 	})
+
+	Context("Legacy Providers", func() {
+		type convertProvidersTableInput struct {
+			legacyProvider    LegacyProvider
+			expectedProviders Providers
+			errMsg            string
+		}
+
+		// Non defaults for these options
+		clientID := "abcd"
+
+		defaultProvider := Provider{
+			ProviderID:   "google_" + clientID,
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+		defaultLegacyProvider := LegacyProvider{
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+
+		displayNameProvider := Provider{
+			ProviderID:   "displayName",
+			ProviderName: "displayName",
+			ClientID:     clientID,
+			ProviderType: "google",
+		}
+
+		displayNameLegacyProvider := LegacyProvider{
+			ClientID:     clientID,
+			ProviderName: "displayName",
+			ProviderType: "google",
+		}
+
+		internalConfigProvider := Provider{
+			ProviderID:   "google_" + clientID,
+			ClientID:     clientID,
+			ProviderType: "google",
+			GoogleConfig: GoogleOptions{
+				GoogleAdminEmail:         "email@email.com",
+				GoogleServiceAccountJSON: "test.json",
+				GoogleGroups:             []string{"1", "2"},
+			},
+		}
+
+		internalConfigLegacyProvider := LegacyProvider{
+			ClientID:                 clientID,
+			ProviderType:             "google",
+			GoogleAdminEmail:         "email@email.com",
+			GoogleServiceAccountJSON: "test.json",
+			GoogleGroups:             []string{"1", "2"},
+		}
+		DescribeTable("convertLegacyProviders",
+			func(o *convertProvidersTableInput) {
```

Comment:
Nit: By convention we normally call this `in`, might be nice to be the same as the others
```suggestion
			func(in *convertProvidersTableInput) {
```

Reply:
sure thing, I guess I followed the pattern in `convertLegacyUpstreams` so I will change it as well..